### PR TITLE
Feature/adding and removing watchers

### PR DESCRIPTION
**Pull request by tilt**
https://www.openproject.org/work_packages/13953

### CODE REVIEW

File: lib/api/v3/work_packages/watchers_api.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          params do
+            requires :user_id, desc: 'Id of the user watching the work package'
+          end
+
+          post do
+            authorize(:add_work_package_watchers, context: @work_package.project)
+
+            user = User.find params[:user_id]
+
+            if @work_package.watcher_users.include?(user)
```
Comment (myabc) [inline lib/api/v3/work_packages/watchers_api.rb]: There is a lot of code here that we might want to move elsewhere. In other words, our Grape code should not end up looking like the Rails Controllers we're replacing.

File: spec/api/watcher_resource_spec.rb [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    before do
+      post post_path, user_id: new_watcher.id
+    end
+
+    context 'authorized user' do
+      let(:current_user) { FactoryGirl.create :user,  member_in_project: project, member_through_role: add_watchers_role }
+
+      it 'should respond with 201' do
+        expect(subject.status).to eq(201)
```
Comment (myabc) [inline spec/api/watcher_resource_spec.rb:35]: if you're going to use named subject, I think `response.status` is clearer than `subject.status`.

**Comment by myabc** [comment]
Looks pretty good so far.

File: lib/api/v3/work_packages/watchers_api.rb (hunk 1 above)
Comment (tilt) [inline lib/api/v3/work_packages/watchers_api.rb]: Any suggestions for restructuring it?

Comment (myabc) [inline lib/api/v3/work_packages/watchers_api.rb]: https://www.youtube.com/watch?v=tg5RFeSfBM4 around 45 minutes in.

FWIW DHH doesn't like this approach (calling it ["Test Induced Design Damage"](https://gist.github.com/dhh/4849a20d2ba89b34b201)).

File: lib/api/v3/work_packages/work_package_representer.rb [hunk 3]
```diff
@@ -68,10 +70,27 @@ def initialize(options = {}, *expand)
         link :assignee do
           {
               href: "#{root_url}/api/v3/users/#{represented.work_package.assigned_to.id}",
-              title: "#{represented.work_package.assigned_to.name} - #{represented.work_package.assigned_to.login}"
+              title: "#{represented.work_package.assigned_to.name} - #{represented.work_package.assigned_to.login}",
```
Comment (ulferts) [inline lib/api/v3/work_packages/work_package_representer.rb]: the ", " at the end needs to be removed

Comment (tilt) [inline lib/api/v3/work_packages/work_package_representer.rb]: this was an accident

**Pull request merged**